src/framework/theme/components/icon/icon-libraries.ts

```diff
@@ -127,13 +133,13 @@ export class NbIconLibraries {
       throwWrongPackTypeError(iconsPack.name, iconsPack.type, 'Font');
     }
 
-    const icon = this.getIconFromPack(name, iconsPack);
+    const icon = this.getIconFromPack(name, iconsPack) ?? '';
 
     return {
       name,
       pack: iconsPack.name,
       type: NbIconPackType.FONT,
-      icon: this.createFontIcon(name, icon ? icon : '', iconsPack.params),
+      icon: this.createFontIcon(name, iconsPack.params.ligature ? name : icon, iconsPack.params),
```

Comment:
Let's extract ternary operator expression result into a variable above and use it here.